### DISCUSSION TITLE
Code coverage

### DISCUSSION THREAD

**Discussion by jcubic** [peggyjs/peggy#289] [Ideas]
I wanted to ask about Code Coverage for projects that use Peggy. I use unit tests and I have (after turning off Branch Coverage on Coveralls) 77% of my language code covered by tests.

Do you have any hints on how to increase coverage? How to figure out from grammar what else needs to be tested.

And the second thing it would be a good idea to add [Coveralls](https://coveralls.io/) to your project. It's free for Open Source projects and you can get a cool badge like this:

[![Coverage Status](https://coveralls.io/repos/github/jcubic/gaiman/badge.svg?branch=master)](https://coveralls.io/github/jcubic/gaiman?branch=master)

**Comment by hildjj** [comment]
Once #240 lands, compile your grammar with `-m inline`, and it gets a lot easier to see paths that you haven't tested.  I'll increase coverage of our `parser.pegjs` when it lands.

Recently I've been preferring [codecov](https://about.codecov.io/) to Coveralls, what's your opinion?  I've been meaning to add support, so this ticket will track adding one of those.

**Comment by jcubic** [comment]
I don't know codecov I've only used Coveralls and I'm happy.
